submission/races.go

```diff
@@ -52,12 +52,12 @@ type groupState struct {
 // When some group is complete cancels all requests that are not needed by any
 // group.
 type safeSubmissionState struct {
-	mu          sync.Mutex
-	logToGroups map[string]ctpolicy.GroupSet
-	groupNeeds  map[string]int // number of logs that need to be submitted for each group.
-	minGroups   int            // minimum number of distinct groups that need a log submitted.
+	mu                     sync.Mutex
+	logToGroups            map[string]ctpolicy.GroupSet
+	groupNeeds             map[string]int // number of logs that need to be submitted for each group.
+	maxSubmissionsPerGroup int            // maximum number of logs that can be submitted to a group.
 
-	groups  map[string]bool // set of groups that have a log submitted.
+	groups  map[string]int // number of logs submitted to each group..
```

Comment:
The `groupNeeds` is very similar to `groups`. Are they actually the same thing?

Reply:
They are slightly different. `groups` shows the number of logs submitted to each group while `groupNeeds` shows the *required* number of logs that still need to be submitted.  `groupNeeds` can be simplified, but I think this requires a separate PR. `groupNeeds` was only used for the old Chrome CT policy so using `baseGroup.MinInclusions` for the total number of required submissions seems more appropriate than a map. This is under the assumption that every policy will have a `baseGroup` (which is true at the moment).

A new PR/issue should be created for this change because there are some unit tests that don't contain a `baseGroup` so there will need to be some slight refactoring on them.

Reply:
The change was smaller than expected so I have tacked it on to this PR.

Reply:
nit: There is an extra `.` at the end of the comment.